### ISSUE TITLE
modal message boxes and input auto focus do not work well together when switching pages

### ISSUE THREAD

**Issue by tbl0605** [bootstrap-vue-next/bootstrap-vue-next#2415]
### Describe the bug

Hi,
this bug is a follow-up to conversation #2401 with @xvaara, there is still an issue where `await new Promise((r) => window.setTimeout(r, 60));` is needed after the modal message box is closed.
For example, when using a modal message box just before leaving a page (using vue-router), autofocus on the next page will not work (without hacks). I have other cases where focus fields don't work after closing a modal message box, but I think it's the same issue.

### Reproduction
- Go to https://stackblitz.com/edit/sb1-jjchwv
- Click on the "Go to Page Two (autofocus OK)" button, the "Page Two" appears and input auto focus works as expected
- Click on the "Go to Page One (with confirmation and delay - autofocus OK)" button, a modal message box appears, click on button "OK", the "Page One" appears and input auto focus works as expected, because `await new Promise((r) => window.setTimeout(r, 60));` is called before switching page
- Click on the "Go to Page Two (with confirmation and no delay - autofocus KO)" button, a modal message box appears, click on button "OK", the "Page Two" appears and input auto focus doesn't work, because there was no delay before switching page

### Reproduction

https://stackblitz.com/edit/sb1-jjchwv

### Used Package Manager

pnpm

**Comment by tbl0605** [comment]
I fixed the access visibility to https://stackblitz.com/edit/sb1-jjchwv, sorry about that.

**Comment by xvaara** [comment]
Yes, as previously stated we (in my view) resolve the promise too soon, on hide, not hidden, so we are still animating away the modals, so it's active still (for 0.3s, or configured fade timing) until the animation is ended. I think we could release the focustrap at the beginning of the animation, when leaving. 

Thanks for the test case, I'll look into this. I'm going through the composables anyway, and making changes at the moment, so big PR coming hopefully this week.

**Comment by xvaara** [comment]
I just noticed that this is because focusTrap, works with no-trap prop set. 

In useActivatedFocusTrap.ts (line 66) when isActive changes there is `await nextTick()`. @VividLemon any idea why we wait for nextTick here? Maybe we should await only then activating, not when deactivating?

**Comment by xvaara** [comment]
@VividLemon you added the await nextTick() in https://github.com/bootstrap-vue-next/bootstrap-vue-next/pull/2068/files#diff-fb1dc9aa3eb4c4a323688f25d44238382884f10dda4b4d88ae7d8a58c8a5e88bR23. It doesn't seem to be needed anymore.
